pandas/tests/test_frame.py

```diff
@@ -3568,6 +3565,20 @@ def test_constructor_tuples(self):
         expected = DataFrame({'A': Series([(1, 2), (3, 4)])})
         assert_frame_equal(result, expected)
 
+    def test_constructor_namedtuples(self):
+        # GH11181
+        from collections import namedtuple
+        named_tuple = namedtuple("Pandas", list('ab'))
+        tuples = [named_tuple(1, 3), named_tuple(2, 4)]
+        expected = DataFrame({'a': [1, 2], 'b': [3, 4]})
+        result = DataFrame(tuples)
+        assert_frame_equal(result, expected)
+
```

Comment:
No, you get `NaN`s, which IIRC is what always happens if you pass lists in with unequal lengths

```python
In [2]:

pd.DataFrame([(1,2,3),(4,5)])
pd.DataFrame([(1,2,3),(4,5)])
Out[2]:
0	1	2
0	1	2	3
1	4	5	NaN
```

If you pass columns, they need to be the max length:
```python
In [4]:

pd.DataFrame([(1,2),(3,4,5)], columns=['a','b'])
pd.DataFrame([(1,2),(3,4,5)], columns=['a','b'])
---------------------------------------------------------------------------
AssertionError                            Traceback (most recent call last)
<ipython-input-4-e2f410a01184> in <module>()
----> 1 pd.DataFrame([(1,2),(3,4,5)], columns=['a','b'])

/usr/local/lib/python2.7/dist-packages/pandas/core/frame.pyc in __init__(self, data, index, columns, dtype, copy)
    262             if len(data) > 0:
    263                 if is_list_like(data[0]) and getattr(data[0], 'ndim', 1) == 1:
--> 264                     arrays, columns = _to_arrays(data, columns, dtype=dtype)
    265                     columns = _ensure_index(columns)
    266 

/usr/local/lib/python2.7/dist-packages/pandas/core/frame.pyc in _to_arrays(data, columns, coerce_float, dtype)
   5211     if isinstance(data[0], (list, tuple)):
   5212         return _list_to_arrays(data, columns, coerce_float=coerce_float,
-> 5213                                dtype=dtype)
   5214     elif isinstance(data[0], collections.Mapping):
   5215         return _list_of_dict_to_arrays(data, columns,

/usr/local/lib/python2.7/dist-packages/pandas/core/frame.pyc in _list_to_arrays(data, columns, coerce_float, dtype)
   5294         content = list(lib.to_object_array(data).T)
   5295     return _convert_object_array(content, columns, dtype=dtype,
-> 5296                                  coerce_float=coerce_float)
   5297 
   5298 

/usr/local/lib/python2.7/dist-packages/pandas/core/frame.pyc in _convert_object_array(content, columns, coerce_float, dtype)
   5352             # caller's responsibility to check for this...
   5353             raise AssertionError('%d columns passed, passed data had %s '
-> 5354                                  'columns' % (len(columns), len(content)))
   5355 
   5356     # provide soft conversion of object dtypes

AssertionError: 2 columns passed, passed data had 3 columns
```